### PR TITLE
coverage: fill some coverage holes

### PR DESCRIPTION
**Pull request by oliver-sanders**
Had a go at filling in some coverage holes whilst ill last week and struggling to work on anything else:

* [refactor] Move "cylc list" specific functions out of the config module and into the script module.
* [test] Test the "cylc list" command.
* [fix] Fix some uncaught "cylc list" option combinations.
* [refactor] Make the unused PointBase.standardise method raise NotImplementedError (both implementations override this method).
* [test] Add test for invalid start task detection in the config module.
* [test] Address some coverage holes in the cycling modules.
* [refactor] Remove unused `is_null` interface (separate commit).

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` (and `conda-environment.yml` if present).
- [x] Tests are included (or explain why tests are not needed).
- [x] Changelog entry included if this is a change that can affect users
- [x] [Cylc-Doc](https://github.com/cylc/cylc-doc) pull request opened if required at cylc/cylc-doc/pull/XXXX.
- [x] If this is a bug fix, PR should be raised against the relevant `?.?.x` branch.

### CODE REVIEW

File: cylc/flow/config.py [hunk 1]
```diff
@@ -1604,18 +1603,6 @@ def get_linearized_ancestors(self):
     def get_first_parent_descendants(self):
         return self.runtime['first-parent descendants']
 
-    @staticmethod
-    def define_inheritance_tree(tree, hierarchy):
```
Comment (oliver-sanders) [inline cylc/flow/config.py:1608]: Moved into `cylc.flow.scripts.list` without changes.

File: cylc/flow/config.py [hunk 2]
```diff
@@ -1651,33 +1638,7 @@ def get_mro(self, ns):
             mro = ["no such namespace: " + ns]
         return mro
 
-    def print_first_parent_tree(self, pretty=False, titles=False):
```
Comment (oliver-sanders) [inline cylc/flow/config.py:1654]: Moved into cylc.flow.scripts.list without changes.

File: cylc/flow/cycling/__init__.py [hunk 3]
```diff
@@ -100,7 +100,7 @@ def standardise(self, allow_truncated: bool = True) -> 'PointBase':
                missing off the front) will be tollerated, if False, truncated
                points will cause an exception to be raised.
         """
-        return self
+        raise NotImplementedError
```
Comment (oliver-sanders) [inline cylc/flow/cycling/__init__.py:103]: No functional change as this method is overridden by both implementations.

File: cylc/flow/cycling/__init__.py [hunk 4]
```diff
@@ -254,9 +254,6 @@ def sub(self, other):
         """Subtract other (interval) from self; return an interval."""
         pass
 
-    def is_null(self):
-        return (self == self.get_null())
```
Comment (oliver-sanders) [inline cylc/flow/cycling/__init__.py:258]: Unused.

File: cylc/flow/scripts/list.py [hunk 5]
```diff
@@ -121,7 +165,14 @@ async def _main(parser: COP, options: 'Values', workflow_id: str) -> None:
     template_vars = get_template_vars(options)
 
     if options.all_tasks and options.all_namespaces:
-        parser.error("Choose either -a or -n")
+        raise InputError("Choose either -a or -n")
+    if (options.all_tasks or options.all_namespaces) and options.prange:
+        raise InputError(
+            '--points cannot be used with --all-tasks or --all-namespaces'
+        )
```
Comment (oliver-sanders) [inline cylc/flow/scripts/list.py:172]: Previously uncaught option combination issue.

File: tests/integration/scripts/test_list.py [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+        'b1',
+        'root',
+    ]
+
+    with pytest.raises(InputError):
+        await cylc_list(capsys, all_tasks=True, all_namespaces=True)
+
+
+async def test_mro(cylc_list, supports_utf8, capsys):
+    """Test the --mro option."""
```
Comment (wxtim) [inline tests/integration/scripts/test_list.py:119]: I really don't like alphabetisms as options.

```suggestion
    """Test the --mro (method resolution order) option."""
```

File: tests/unit/cycling/test_integer.py [hunk 7]
```diff
@@ -219,10 +224,59 @@ def test_interval_arithmetic():
...
+    assert IntegerPoint(1) <= IntegerPoint(2)
+    assert IntegerPoint(2) >= IntegerPoint(1)
+    assert IntegerPoint(1) != IntegerPoint(2)
+    assert IntegerInterval('P1') == IntegerInterval('P1')
+    assert IntegerInterval('P1') < IntegerInterval('P2')
+    assert IntegerInterval('P1') <= IntegerInterval('P2')
+    assert IntegerInterval('P2') >= IntegerInterval('P1')
+    assert IntegerInterval('P1') != IntegerInterval('P2')
+
+    # None comparisons work counter intuatively
```
Comment (wxtim) [inline tests/unit/cycling/test_integer.py:261]:
```suggestion
    # None comparisons work counter intuitively
```

**Review by wxtim — APPROVED** [review]
Looks reasonable. I found a single spelling error and had a grumble about acronyms. Nothing actually significant to report.

File: tests/integration/scripts/test_list.py (hunk 6 above)
Comment (hjoliver) [reply tests/integration/scripts/test_list.py:119]: Point taken, but it's pretty niche and the long form is VERY long for an option name, so I went for explaining in the help text:
```
  -m, --mro             Print the linear "method resolution order" for each
                        namespace (the multiple-inheritance precedence order
                        as determined by the C3 linearization algorithm).
```
But feel free to put up a quick PR for `--method-resolution-order` if you like (not sure I'd bother, but no harm in having it).

**Pull request merged**